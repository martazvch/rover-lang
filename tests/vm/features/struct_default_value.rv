struct Vec2 {
    a, b: float,
    x: int = 8,
    y: bool,
    z, w: str = "yup",

    fn display(self) {
        print self.a
        print self.b
        print self.x
        print self.y
        print self.z
        print self.w
    }
}


fn main() {
    var v = Vec2 { a = 1, b = 2, y = true }
    v.display()


    v = Vec2 { y = false, x = 7, a = 3, w = "Tom",  b = 99 }
    v.display()
}
