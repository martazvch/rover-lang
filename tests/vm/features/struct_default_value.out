1.2
2.4
8
true
yup
yup
false
false

3.3
99.99
7
false
yup
Tom
false
false
